### PR TITLE
Use sa token for metrics http autodiscovery and pyroscope pprof scraping

### PR DESCRIPTION
**Pull request by jingkaihe**
This allows alloy to scrape endpoint secured by some rbac termination endpoint such as [kube-rbac-proxy](https://github.com/brancz/kube-rbac-proxy).

This is especially beneficial to the pyroscope pprof scraping, otherwise we will end up with exposing `pod_ip:6060` cluster wide. with the kubernetes sa token passed in we can instead:

* run the pyroscope endpoint on `127.0.0.1:6060` which is only accessible within the pod itself.
* kube-rbac-proxy authn and authz with token review and subject access review (non resource url `/debug/pprof/*`) and forward the request in the manner of `0.0.0.0:6062 -> 127.0.0.1:6060`
* on the profiler end we can thus grant the sa with non resource url `/debug/pprof/*

end result:

```
root@grafana-k8s-monitoring-alloy-profiles-4lv2x:/# curl "http://11.5.6.27:6062/debug/pprof/"
Unauthorized
root@grafana-k8s-monitoring-alloy-profiles-4lv2x:/# curl -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" "http://11.5.6.27:6062/debug/pprof/"
<html>
<head>
<title>/debug/pprof/</title>
<style>
.profile-name{
        display:inline-block;
```

Please let me know what you think :)

### CODE REVIEW

File: charts/k8s-monitoring/Chart.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ apiVersion: v2
 name: k8s-monitoring
 description: A Helm chart for gathering, scraping, and forwarding Kubernetes telemetry data to a Grafana Stack.
 type: application
-version: 1.1.1
+version: 1.1.2
```
Comment (petewall) [inline charts/k8s-monitoring/Chart.yaml:6]: Can you change this back? We control the versioning when its time to release"

**Review by petewall — COMMENTED** [review]
Let me try this out and let you know.

File: charts/k8s-monitoring/Chart.yaml (hunk 1 above)
Comment (jingkaihe) [reply charts/k8s-monitoring/Chart.yaml]: Done at https://github.com/grafana/k8s-monitoring-helm/pull/580/commits/3406dfc64dd8c564986465841cd57a97135899b2

**Comment by jingkaihe** [comment]
@petewall thanks. We've been running this in production for approx. a week now.

this is the annotation we use

```yaml
  annotations:
    profiles.grafana.com/block.path: /debug/pprof/block
    profiles.grafana.com/block.port_name: pprof
    profiles.grafana.com/block.scheme: http
    profiles.grafana.com/block.scrape: "true"
    profiles.grafana.com/cpu.path: /debug/pprof/profile
    profiles.grafana.com/cpu.port_name: pprof
    profiles.grafana.com/cpu.scheme: http
    profiles.grafana.com/cpu.scrape: "true"
    profiles.grafana.com/fgprof.scrape: "false"
    profiles.grafana.com/goroutine.path: /debug/pprof/goroutine
    profiles.grafana.com/goroutine.port_name: pprof
    profiles.grafana.com/goroutine.scheme: http
    profiles.grafana.com/goroutine.scrape: "true"
    profiles.grafana.com/memory.path: /debug/pprof/heap
    profiles.grafana.com/memory.port_name: pprof
    profiles.grafana.com/memory.scheme: http
    profiles.grafana.com/memory.scrape: "true"
    profiles.grafana.com/mutex.path: /debug/pprof/mutex
    profiles.grafana.com/mutex.port_name: pprof
    profiles.grafana.com/mutex.scheme: http
    profiles.grafana.com/mutex.scrape: "true"
```

this is the sidecar we use

```yaml
  containers:
  - args:
    - --insecure-listen-address=0.0.0.0:6062
    - --upstream=http://127.0.0.1:6060
    image: quay.io/brancz/kube-rbac-proxy:v0.18.0
    name: profiles-rbac-proxy
    ports:
    - containerPort: 6062
      name: pprof
      protocol: TCP
```

un-secured pprof endpoint runs on 127.0.0.1 so it's not accessible to other pods

rbac config more or less looks like https://github.com/brancz/kube-rbac-proxy/tree/master/examples/non-resource-url

**Review by petewall — APPROVED** [review]

**Pull request merged**